web-common/src/features/dashboards/pivot/pivot-expansion.ts

```diff
@@ -484,22 +516,37 @@ export function addExpandedDataToPivot(
         );
       }
 
-      parent[lastIdx].subRows = subTableData?.map((row) => {
-        const newRow = {
-          ...row,
-          [rowDimensions[0]]: row[anchorDimension],
-        };
-
-        /**
-         * Add sub rows to the new row if number of row dimensions
-         * is greater than number of nest levels expanded except
-         * for the last level
-         */
-        if (numRowDimensions - 1 > indices.length) {
-          newRow.subRows = [{ [rowDimensions[0]]: "LOADING_CELL" }];
-        }
-        return newRow;
-      });
+      const mappedSubRows =
+        subTableData?.map((row) => {
+          const newRow = {
+            ...row,
+            [rowDimensions[0]]: row[anchorDimension],
+          };
+
+          /**
+           * Add sub rows to the new row if number of row dimensions
+           * is greater than number of nest levels expanded except
+           * for the last level
+           */
+          if (numRowDimensions - 1 > indices.length) {
+            newRow.subRows = [{ [rowDimensions[0]]: "LOADING_CELL" }];
+          }
+          return newRow;
+        }) || [];
+
+      // Add "Show More" button row if more data is available and we're not at max limit (100)
+      if (
+        expandedRowData.hasMore &&
+        expandedRowData.effectiveLimit &&
+        expandedRowData.effectiveLimit < 100
+      ) {
+        mappedSubRows.push({
+          [rowDimensions[0]]: SHOW_MORE_BUTTON,
```

Comment:
Not too comfortable using different value for types. If we have to use it then make it a bit more obscure perhaps? Use `__rill_type` as prefix or something.